### DISCUSSION TITLE
Metaframework/SSR/Isomorphic environment.

### DISCUSSION THREAD

**Discussion by Blankeos** [maxnowack/signaldb#790] [Q&A]
For apps that usually work with isomorphic UIs, do you have an example? NextJS or SvelteKit, Solid Start, etc.

I assume SignalDB only has to run on the client. Never on the server. So I think just plastering `export const collection = new Collection(...)` in a plain file wouldn't really fly well if you `import { collection } from "./"`. I think it has to be dynamically imported, or instantiated once the app mounts on the client.

I managed to get RxDB to work in an SSR environment by using Context and some listeners: https://github.com/Blankeos/rxdb-solid.

Since SignalDB has reactivity adapters, I'm wondering if I can just use that instead of manually attaching listeners. So far, I couldn't get it to work tho.

**Comment by maxnowack** [comment]
I don’t have an example for usage in an isomorphic environment yet. SignalDB should work on the server side as well. Exporting the collection in one file and importing it from both the server and client should work, as the collection itself doesn’t have any state or data directly after initialization. I took a look at your RxDB example. I think it should work quite similar if you're using a context for the SignalDB collection like you've done it [here](https://github.com/Blankeos/rxdb-solid/blob/main/stores/database.store.tsx) for RxDB.

To make sense in a scenario with SSR and data replication, we need to transfer the data somehow from the server to the client, so that requests to the backend are not made twice. I'm not familiar with transferring state in a SSR environment to the client. I would need help for implementing this.
